Develop 2025.1.0.0-preview Support Spring Boot 4.0-closed

yingqiaoli8 opened the pull request:
### Describe what this PR does / why we need it
Project Revision: 2025.1.0.0-preview
Spring Boot version: 4.0.0
Spring Cloud version: 2025.1.0
spring Cloud build: 5.0.0

### Does this pull request fix one issue?

<!--If that, add "Fixes #xxxx" below in the next line. For example, Fixes #15. Otherwise, add "NONE" -->

### Describe how you did it
1. update the pom.xml and make it support Spring Boot 4.0
2. update the readme file
3. update the code to support new version dependency
4. org.springframework.boot.env.EnvironmentPostProcessor change to org.springframework.boot.EnvironmentPostProcessor and avoid  SpringFactoriesEnvironmentPostProcessorsFactory
<img width="1823" height="909" alt="image" src="https://github.com/user-attachments/assets/09ef0e26-b6f0-498b-a7e3-fde2e152a8ce" />
<img width="945" height="657" alt="image" src="https://github.com/user-attachments/assets/64d9b7bd-0199-443a-b76c-43c61952b9e1" />
5. RocketMQInboundChannelAdapter change the retry logic (based on **RequestHandlerRetryAdvice.**class)
<img width="1803" height="900" alt="image" src="https://github.com/user-attachments/assets/f998cbbf-7b4f-4846-b6aa-577fa4b9cb95" />
<img width="1797" height="858" alt="image" src="https://github.com/user-attachments/assets/5c0d53b9-5880-404f-bd1e-a8e751b7e2ac" />


### Describe how to verify it
1. maven package it, make sure compile successfully;
2. run the test file with every module;
3. maven install it and change your other old project to the install version.

### Special notes for reviews

1. try skip the checkstyle and test when you package your code and open them when you debug your code
2. Prepare a rollback plan before deploy the new version code

liyingqiao121727 committed:
feat: 升级pom文件springboot，springcloud 相关的版本号

liyingqiao121727 committed:
feat: 更新 pom 依赖

liyingqiao121727 committed:
chore: 更新readme

liyingqiao121727 committed:
chore: update GatewayConfig example

liyingqiao121727 committed:
feat: update Test class

liyingqiao121727 committed:
feat: update schedulerx class

liyingqiao121727 committed:
feat: update sidecar class

liyingqiao121727 committed:
feat: update seata class

liyingqiao121727 committed:
feat: update circuitbreaker-sentinel class

liyingqiao121727 committed:
feat: update nacos-discovery class

liyingqiao121727 committed:
feat: update alibaba-sentinel class

liyingqiao121727 committed:
feat: update nacos-config class

liyingqiao121727 committed:
feat: update rocketmq class

liyingqiao121727 committed:
feat: update org.springframework.boot.env.EnvironmentPostProcessor to org.springframework.boot.EnvironmentPostProcessor, avoid  SpringFactoriesEnvironmentPostProcessorsFactory

liyingqiao121727 committed:
feat: update and implement RocketMQInboundChannelAdapter.class' retry logic

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/alibaba/spring-cloud-alibaba?pullRequest=4157) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/alibaba/spring-cloud-alibaba?pullRequest=4157) before we can accept your contribution.<br/>**4** out of **5** committers have signed the CLA.<br/><br/>:white_check_mark: yingqiaoli8<br/>:white_check_mark: xuxiaowei-com-cn<br/>:white_check_mark: uuuyuqi<br/>:white_check_mark: yuhuangbin<br/>:x: liyingqiao121727<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/alibaba/spring-cloud-alibaba?pullRequest=4157) it.</sub>

liyingqiao121727 committed:
feat: update import Order and pom webmvc to webflux and EnvironmentPostProcessorApplicationListener import modify

liyingqiao121727 committed:
feat: code modify

liyingqiao121727 committed:
chore: update checkstyle

liyingqiao121727 committed:
feat: update Test class

liyingqiao121727 committed:
feat: update Test class again

liyingqiao121727 committed:
Merge branch '2025.1.x' into develop-2025.1.0.0

# Conflicts:
#	pom.xml
#	spring-cloud-alibaba-dependencies/pom.xml
#	spring-cloud-alibaba-examples/integrated-example/integrated-gateway/src/main/java/com/alibaba/cloud/integration/gateway/config/GatewayConfig.java
#	spring-cloud-alibaba-examples/nacos-example/readme-zh.md
#	spring-cloud-alibaba-examples/nacos-example/readme.md
#	spring-cloud-alibaba-starters/spring-alibaba-nacos-config/src/main/java/com/alibaba/cloud/nacos/configdata/NacosConfigDataLocationResolver.java
#	spring-cloud-alibaba-starters/spring-alibaba-nacos-config/src/test/java/com.alibaba.cloud.nacos/endpoint/NacosConfigEndpointTests.java
#	spring-cloud-alibaba-starters/spring-cloud-circuitbreaker-sentinel/src/main/java/com/alibaba/cloud/circuitbreaker/sentinel/feign/CircuitBreakerRuleChangeListener.java
#	spring-cloud-alibaba-starters/spring-cloud-circuitbreaker-sentinel/src/main/java/com/alibaba/cloud/circuitbreaker/sentinel/feign/SentinelFeignClientProperties.java
#	spring-cloud-alibaba-starters/spring-cloud-circuitbreaker-sentinel/src/test/java/com/alibaba/cloud/circuitbreaker/sentinel/SentinelCircuitBreakerIntegrationTest.java
#	spring-cloud-alibaba-starters/spring-cloud-starter-alibaba-nacos-discovery/src/main/java/com/alibaba/cloud/nacos/registry/NacosAutoServiceRegistration.java
#	spring-cloud-alibaba-starters/spring-cloud-starter-alibaba-nacos-discovery/src/main/java/com/alibaba/cloud/nacos/registry/NacosServiceRegistryAutoConfiguration.java
#	spring-cloud-alibaba-starters/spring-cloud-starter-alibaba-nacos-discovery/src/test/java/com/alibaba/cloud/nacos/discovery/NacosDiscoveryLoadBalancerConfigurationTest.java
#	spring-cloud-alibaba-starters/spring-cloud-starter-alibaba-seata/pom.xml
#	spring-cloud-alibaba-starters/spring-cloud-starter-alibaba-seata/src/main/java/com/alibaba/cloud/seata/webclient/SeataWebClientAutoConfiguration.java
#	spring-cloud-alibaba-starters/spring-cloud-starter-alibaba-seata/src/main/java/com/alibaba/cloud/seata/webclient/SeataWebClientBuilderCustomizer.java
#	spring-cloud-alibaba-starters/spring-cloud-starter-alibaba-sentinel/pom.xml
#	spring-cloud-alibaba-starters/spring-cloud-starter-alibaba-sentinel/src/main/java/com/alibaba/cloud/sentinel/rest/SentinelClientHttpResponse.java
#	spring-cloud-alibaba-starters/spring-cloud-starter-stream-rocketmq/src/main/java/com/alibaba/cloud/stream/binder/rocketmq/integration/inbound/RocketMQInboundChannelAdapter.java
#	spring-cloud-alibaba-starters/spring-cloud-starter-stream-rocketmq/src/main/java/com/alibaba/cloud/stream/binder/rocketmq/support/JacksonRocketMQHeaderMapper.java
#	spring-cloud-alibaba-tests/sentinel-tests/sentinel-degrade-test/src/test/java/com/alibaba/cloud/tests/sentinel/degrade/SentinelDegradeTestAppTest.java
#	spring-cloud-alibaba-tests/sentinel-tests/sentinel-flowcontrol-test/src/test/java/com/alibaba/cloud/tests/sentinel/degrade/SentinelFlowControlTestAppTest.java

liyingqiao121727 committed:
feat: update Test class and pom

liyingqiao121727 committed:
feat: revert code

liyingqiao121727 committed:
feat: modify code

liyingqiao121727 committed:
feat: modify code

liyingqiao121727 committed:
feat: readme

liyingqiao121727 committed:
feat: readme

xuxiaowei-com-cn committed:
chore: restore retry mechanism for message sending in RocketMQInboundChannelAdapter (#4162)

chore: restore retry mechanism for message sending in RocketMQInboundChannelAdapter (#4162)

xuxiaowei-com-cn committed:
:bug: update NacosConfigDataLocationResolver registerConfigManager method (#4164)

liyingqiao121727 committed:
Merge branch '2025.1.x' into develop-2025.1.0.0-preview

# Conflicts:
#	spring-cloud-alibaba-starters/spring-cloud-starter-alibaba-seata/pom.xml
#	spring-cloud-alibaba-starters/spring-cloud-starter-alibaba-seata/src/main/java/com/alibaba/cloud/seata/webclient/SeataWebClientBuilderCustomizer.java
#	spring-cloud-alibaba-starters/spring-cloud-starter-stream-rocketmq/src/main/java/com/alibaba/cloud/stream/binder/rocketmq/integration/inbound/RocketMQInboundChannelAdapter.java

uuuyuqi committed:
Revert "Fixbug issue4147 2025.1.x，同 PR 4148" (#4165)

xuxiaowei-com-cn committed:
chore: add actions to release github package (#4188)

chore: add actions to release github package (#4188)

xuxiaowei-com-cn committed:
chore: upgrade nacos client from 3.0.3 to 3.1.0 (#4193)

chore: upgrade nacos client from 3.0.3 to 3.1.0 (#4193)

yuhuangbin committed:
code fix: remove bootstrap config support, instead spring.config.import (#4167)

yingqiaoli8 committed:
Merge branch '2025.1.x' into develop-2025.1.0.0-preview

yingqiaoli8 committed:
Merge remote-tracking branch 'origin/develop-2025.1.0.0-preview' into develop-2025.1.0.0-preview
